dataclasses_avroschema/model_generator/lang/python/base.py

```diff
@@ -448,7 +460,12 @@ def parse_enum(self, field: JsonDict) -> str:
 
         docstring = self.render_docstring(docstring=field.get("doc"))
         enum_class = templates.enum_template.safe_substitute(name=enum_name, symbols=symbols_repr, docstring=docstring)
-        metaclass = self.render_metaclass(schema=field, decorator=templates.METACLASS_DECORATOR)
+        add_schema_name = not enum_name == field["name"]
```

Comment:
You can do: `add_schema_name = enum_name != field["name"]`

Reply:
Agreed!